indico/modules/events/registration/fields/simple.py

```diff
@@ -374,3 +375,16 @@ def _indico_email(value):
                 raise ValidationError(_('Invalid email address'))
 
         return _indico_email
+
+
+class CaptchaField(fields.String):
+    """Validates an answer against a CAPTCHA stored in the session."""
+
+    def __init__(self):
+        super().__init__(required=True, validate=not_empty)
+
+    def _validate(self, user_answer):
+        super()._validate(user_answer)
+        answer = session.get('captcha_answer', None)
+        if not answer or answer != user_answer:
```

Comment:
This needs to be a plugin method - in case of reCaptcha or any of the other "fancy" ones (where you have to align a puzzle tile or do similar things) you typically need to call an external API to verify the result.